src/SrvBroker.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#ifdef INPUTSTREAM_TEST_BUILD
+#include "test/KodiStubs.h"
+#else
+#include <kodi/AddonBase.h>
+#endif
+
+#include <map>
+#include <memory>
+#include <string>
+
+// forward
+namespace ADP::SETTINGS
+{
+class CCompSettings;
+}
+namespace ADP::KODI_PROPS
+{
+class CCompKodiProps;
+}
+
+/*
+ * \brief Service broker is a singleton that give an easy access to the resources from anywhere in the code.
+ */
+class ATTR_DLL_LOCAL CSrvBroker
+{
+public:
+  CSrvBroker(CSrvBroker& other) = delete; // Not clonable
+  void operator=(const CSrvBroker&) = delete; // Not assignable
+
+  /*
+   * \brief Initialize service broker components.
+   */
+  void Init(const std::map<std::string, std::string>& kodiProps);
+
+  static CSrvBroker* GetInstance()
+  {
+    static CSrvBroker instance;
+    return &instance;
+  }
+
+  /*
+   * \brief Get Kodi properties component, to read Kodi (Listitem / playlist files "STRM") properties.
+   */
+  static std::shared_ptr<ADP::KODI_PROPS::CCompKodiProps> GetKodiProps()
```

Comment:
Why not a unique_ptr and return a refence? Do you actually need a shared_ptr?

Reply:
yeah i think we can also do in that way

Reply:
i tried but looks like its not possible to do
seem that unique_ptr require class to be declared and not foward declared as done here

Reply:
> i tried but looks like its not possible to do
> 
> seem that unique_ptr require class to be declared and not foward declared as done here

You can forward declare a class used in a unique_ptr. If you have compilation issues it may be because you are missing a constructor/destructor which may need to be added to the .cpp file (even if they are just declared default).

Reply:
i have tried as you said by creating constructor/destructor in to cpp for each class but compiler produce same errors
`error C2338: static_assert failed: 'can't delete an incomplete type'`
at this point better use raw pointers instead of smartpointer and create and delete them manually